candle-transformers/src/models/debertav2.rs

```diff
@@ -0,0 +1,1499 @@
+        let relative_attention = config.relative_attention;
+        let mut max_relative_positions = config.max_relative_positions;
+
+        let mut pos_ebd_size: isize = 0;
+        let position_buckets = config.position_buckets.unwrap_or(-1);
+        let mut pos_dropout: Option<StableDropout> = None;
+        let mut pos_key_proj: Option<candle_nn::Linear> = None;
+        let mut pos_query_proj: Option<candle_nn::Linear> = None;
+
+        if relative_attention {
+            if max_relative_positions < 1 {
+                max_relative_positions = config.max_position_embeddings as isize;
+            }
+            pos_ebd_size = max_relative_positions;
+            if position_buckets > 0 {
+                pos_ebd_size = position_buckets
+            }
+
+            pos_dropout = Some(StableDropout::new(config.hidden_dropout_prob));
+
+            if !share_att_key {
+                if config.pos_att_type.contains(&"c2p".to_string()) {
+                    pos_key_proj = Some(candle_nn::linear(
+                        config.hidden_size,
+                        all_head_size,
+                        vb.pp("pos_key_proj"),
+                    )?);
+                }
+                if config.pos_att_type.contains(&"p2c".to_string()) {
+                    pos_query_proj = Some(candle_nn::linear(
+                        config.hidden_size,
+                        all_head_size,
+                        vb.pp("pos_query_proj"),
+                    )?);
+                }
+            }
+        }
+
+        let dropout = StableDropout::new(config.attention_probs_dropout_prob);
+        let device = vb.device().clone();
+
+        Ok(Self {
+            config,
+            num_attention_heads,
+            query_proj,
+            key_proj,
+            value_proj,
+            dropout,
+            device,
+            relative_attention,
+            pos_dropout,
+            position_buckets,
+            max_relative_positions,
+            pos_ebd_size,
+            share_att_key,
+            pos_key_proj,
+            pos_query_proj,
+        })
+    }
+
+    pub fn forward(
+        &self,
+        hidden_states: &Tensor,
+        attention_mask: &Tensor,
+        query_states: Option<&Tensor>,
+        relative_pos: Option<&Tensor>,
+        rel_embeddings: Option<&Tensor>,
+    ) -> candle::Result<Tensor> {
+        let query_states = match query_states {
+            Some(qs) => qs,
+            None => hidden_states,
+        };
+
+        let query_layer = self.transpose_for_scores(&self.query_proj.forward(query_states)?)?;
+        let key_layer = self.transpose_for_scores(&self.key_proj.forward(query_states)?)?;
+        let value_layer = self.transpose_for_scores(&self.value_proj.forward(query_states)?)?;
+
+        let mut rel_att: Option<Tensor> = None;
+
+        let mut scale_factor: usize = 1;
+
+        if self.config.pos_att_type.contains(&"c2p".to_string()) {
```

Comment:
Wouldn't it work to just use `"c2p"` rather than making a string and referencing it?

Reply:
The way I have it set up in the config is that `pos_att_type` is parsed into a `Vec<String>` by basically splitting the string on a pipe character. Or, if it's specified as a JSON array in the config, it will be translated directly into a `Vec<String>`. I addressed this in an update to use an iterator and check for `any()`.